core/timeline_inc.php

```diff
@@ -74,15 +80,15 @@
 				echo '&#160;&#160;';
 
 				echo '<div class="btn-group">';
-				$t_url_params = array( 'days' => $f_days + 7 );
-				$t_href = $t_url_page . '?' . http_build_query( $t_url_params );
+				$t_url_params['days'] = $f_days + 7;
+				$t_href = $t_url_page . '?' . htmlspecialchars( http_build_query( $t_url_params ) );
```

Comment:
We use `http_build_query` in several places without `htmlspecialchars` or any post processing.  If we need to post process, then we should do it in all locations ideally via a Mantis API that is used from all places.  If not, then it should be removed.

Reply:
It uses `htmlspecialchars` becasue it's printing all GET parameters that were present in the request. Those parameters are unverified content, so i think this avoid xss.
Other usages of `http_build_query`, where the parameters are trusted, don't need any postprocessing.

Reply:
@cproensa Makes sense.  But shouldn't the escaping happens on each untrusted value rather than the output of the `http_build_query`?  This would avoid escaping strings added by `http_build_query` itself like `&`s.